lib/plug/ssl.ex

```diff
@@ -43,9 +43,39 @@ defmodule Plug.SSL do
     * `:host` - a new host to redirect to if the request's scheme is `http`,
       defaults to `conn.host`. It may be set to a binary or a tuple
       `{module, function, args}` that will be invoked on demand
+    * `:cipher_suite` - select a preconfigured cipher suite as described below.
+      You can select `strong ` or `compatible`. If an option is not specified it inherits
+      the defaults from Cowboy and Ranch to preserve backwords compatibility.
     * `:log` - The log level at which this plug should log its request info.
       Default is `:info`. Can be `false` to disable logging.
 
+  ## Cipher Suites
+
+  _The cipher suites were last updated on 2018-JUN-14._
+
+  To simplify configuration of TLS defaults Plug provides two preconfifured
+  options: `:strong` and `:compatible`. The Ciphers chosen and related configuration
+  come from the OWASP recommendations found here:
+  https://www.owasp.org/index.php/TLS_Cipher_String_Cheat_Sheet
+
+  The **Strong(()) cipher suite only supports tlsv1.2. Ciphers were based on the OWASP Group A+
+  and includes support for RSA or ECDSA certificates. The intention of this configuration is
+  to provide as secure as possible defaults knowing that it will not be fully compatible with
+  older browsers and operating systems.
+
+  The **Compatible** cipher suite supports tlsv1, tlsv1.1 and tlsv1.2. Ciphers were based on the
+  OWASP Group B and includes support for RSA or ECDSA certificates. The intention of this
+  configuration is to provide as secure as possible defaults that still maintain support for
+  older browsers and Android versions 4.3 and earlier
+
+  For both suites we've specified ceritifcate curves ecp384r1 and secp521r1. We would
+  have included prime256v1 but it is not supported by Erlang. Since OWASP doesn't perscribe
```

Comment:
Ah, right, this is confusing: prime256v1 is just a different name for secp256r1. Erlang supports this curve, but `:ssl` only recognises the SECP naming, while `:crypto` recognises both. We should use `[:secp256r1, :secp384r1, :secp521r1]` for the eccs parameter. Sorry for not spotting this earlier.